DataFormats/Portable/interface/PortableCollection.h

```diff
@@ -47,8 +48,10 @@ using PortableMultiCollection = typename traits::PortableMultiCollectionTrait<TD
 // define how to copy PortableCollection between host and device
 namespace cms::alpakatools {
   template <typename TLayout, typename TDevice>
+    requires alpaka::isDevice<TDevice>
```

Comment:
These `isDevice`/`isQueue` requirements are not strictly required for this PR, but I felt they wouldn't hurt.

I did play with requiring them to be of non-CPU for all cases (as that is more or less the intention), but in the end leaned to go with the actual requirements of the implementation of the `copyAsync()` functions for this PR. We can discuss (here or separately) if making these constraints more strict than minimally required would be useful.